crates/containerd-shim-wasmtime/src/tests.rs

```diff
@@ -101,3 +101,42 @@ fn test_has_default_devices() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+// Test that the shim can execute an named exported function
+// that is not the default _start function in a wasm component.
+// The current limitation is that there is no way to pass arguments
+// to the exported function.
+// Issue that tracks this: https://github.com/containerd/runwasi/issues/414
+#[test]
+#[serial]
+fn test_simple_component() -> anyhow::Result<()> {
+    let (exit_code, _, _) = WasiTest::<WasiInstance>::builder()?
+        .with_wasm(SIMPLE_COMPONENT)?
+        .with_start_fn("thunk")?
+        .build()?
+        .start()?
+        .wait(Duration::from_secs(10))?;
+
+    assert_eq!(exit_code, 0);
+
+    Ok(())
+}
+
+// Test that the shim can execute a wasm component that is
+// compiled with wasip2.
+//
+// This is using the `wasi:cli/command` world to run the component.
+#[test]
+#[serial]
+fn test_wasip2_component() -> anyhow::Result<()> {
+    let (exit_code, stdout, _) = WasiTest::<WasiInstance>::builder()?
+        .with_wasm(COMPONENT_HELLO_WORLD)?
```

Comment:
how was this component compiled?

Reply:
yeah I intended to not make big changes to our exiting test infras as compiling components requires a different set of tools than compiling modules. 

This component was compiled from another place, essentially followed https://component-model.bytecodealliance.org/language-support/rust.html but added the wasip2

Reply:
maybe comment on how to compile it?  I worry about future me trying to figure this out :-)

Reply:
Done.